input/script_utils.go

```diff
@@ -638,6 +639,12 @@ type HtlcScriptTree struct {
 	// TimeoutTapLeaf is the tapleaf for the timeout path.
 	TimeoutTapLeaf txscript.TapLeaf
 
+	// AuxLeaf is an axuillaruy leaf that can be used to extend the base
+	// HTLC script tree with new spend paths, or just as extra commitment
+	// space. When present, this leaf will always be in the left-most or
+	// right-most area of the tapscript tree.
+	AuxLeaf fn.Option[txscript.TapLeaf]
```

Comment:
in order to be open-ended with regards to flexibility, couldn't we have multiple aux leaves here? something like `fn.Slice`

we could hide everything under a single `AuxLeaf` but I think just having multiple leaves will keep it simpler overall on the script level itself